tvb/analyzers/fragility_lib/execute/singlenode/parallelmvar.py

```diff
@@ -0,0 +1,136 @@
+# Parallel Processing Functions
+# Authors: Adam Li
+# Edited by: Adam Li
+
+# Imports necessary for this function 
+import numpy as np 
+from ...linearmodels.basemodel import BaseWindowModel
+from ...linearmodels.mvarmodel import MvarModel
+
+# for multiprocessing on separate cpus
+import psutil
+import multiprocessing as mp
+
+import functools
+from contextlib import closing
+
+# utility libraries
+import os
+import traceback
+import natsort
+import sys
+import warnings
+
+import tempfile
+
+from tvb.basic.logger.builder import get_logger
+LOG = get_logger(__name__)
+
+def _initmvar(raweeg, timepoints, mvarmodel, tempdir):
+    global shared_raweeg
+    global shared_timepoints
+    global shared_mvarmodel
+    global shared_tempdir
+
+    shared_timepoints = timepoints
+    shared_raweeg = raweeg
+    shared_mvarmodel = mvarmodel        
+    shared_tempdir = tempdir
+
+def mvarjob(icore):
+    winstoanalyze = processlist[icore]
+    for win in winstoanalyze:
+        # 1: fill matrix of all channels' next EEG data over window
+        eegwin = shared_raweeg[:, (shared_timepoints[win,0]) : (shared_timepoints[win,1]+1)]
+
+        # 2: call the function to create the mvar model
+        adjmat = shared_mvarmodel.mvaradjacencymatrix(eegwin)
+        
+        tempfilename = os.path.join(shared_tempdir, 'temp_'+str(win)+'.npz')
+        try:
+            np.savez_compressed(tempfilename, adjmat=adjmat)
+        except:
+            sys.stdout.write(traceback.format_exc())
+            return 0
+        # print("finished mvar job at ", win)
+        # sys.stdout.flush()
+    return 1
+
+def mvarjobwin(win):
+    # 1: fill matrix of all channels' next EEG data over window
+    eegwin = shared_raweeg[:, (shared_timepoints[win,0]) : (shared_timepoints[win,1]+1)]
+    # normalize eeg
+    # eegwin = _normalizets(eegwin)
+    # eegwin = (eegwin - np.mean(eegwin, axis=1)[:, np.newaxis]) / np.std(eegwin, axis=1)[:, np.newaxis]
+
+    # 2: call the function to create the mvar model
+    adjmat = shared_mvarmodel.mvaradjacencymatrix(eegwin)
+        
+    tempfilename = os.path.join(shared_tempdir, 'temp_'+str(win)+'.npz')
+    try:
+        np.savez_compressed(tempfilename, adjmat=adjmat)
+    except:
+        sys.stdout.write(traceback.format_exc())
+        return 0
+    return 1
+
+class ParallelMvar(BaseWindowModel):
+    def __init__(self, winsizems=250, stepsizems=125, samplerate=1000,numcores=None):
+        BaseWindowModel.__init__(self, 
```

Comment:
`super(ParallelMvar, self).__init__(...)` is idiomatic because it handles multiple inheritance correctly.

Reply:
Updated to be super(<classname>, self) for each file in the execute dir.